src/main/java/frc/robot/subsystems/shooter/FlywheelIOTalonFX.java

```diff
@@ -0,0 +1,103 @@
+    flywheelConfig.CurrentLimits.StatorCurrentLimit = FlywheelConstants.FLYWHEEL_STATOR_LIMIT;
+    flywheelConfig.CurrentLimits.SupplyCurrentLimitEnable = FlywheelConstants.FLYWHEEL_SUPPLY_ENABLE;
+    flywheelConfig.CurrentLimits.StatorCurrentLimitEnable = FlywheelConstants.FLYWHEEL_STATOR_ENABLE;
+    flywheelConfig.MotorOutput.NeutralMode = NeutralModeValue.Coast; //This is used to ensure maximum power efficiency, to ensure flywheels keep spinning after power off.
+ 
+    leaderFlywheelMotor.getConfigurator().apply(flywheelConfig); //Apply the flywheel config defined above
+    flywheelConfig.MotorOutput.Inverted = InvertedValue.Clockwise_Positive; //invert one so that they can go in the same direction
+    followerFlywheelMotor.getConfigurator().apply(flywheelConfig);
+    followerFlywheelMotor.setControl(new Follower(leaderFlywheelMotor.getDeviceID(), false)); //phoenix 6 method to make the follower motor to copy, so code only needed to write for leader. NOT NECESSARY, JUST CONVENTION STOLEN EASTBOT
+
+    BaseStatusSignal.setUpdateFrequencyForAll( //sets the frequency where the values are updated
+        250.0, leaderPosition, leaderVelocity, leaderAppliedVolts, leaderCurrent, followerCurrent);
+    leaderFlywheelMotor.optimizeBusUtilization(); //updates frequency based on how often the bus communicates with the motor
+    followerFlywheelMotor.optimizeBusUtilization();
+  }
+
+  @Override
+  public void updateInputs(FlywheelIOInputs inputs) {//gets current values for motors and puts them into a log
+    BaseStatusSignal.refreshAll(
+        leaderPosition, leaderVelocity, leaderAppliedVolts, leaderCurrent, followerCurrent);
+    inputs.positionRotaions = leaderPosition.getValueAsDouble() / GEAR_RATIO; //converted to radians to gear ratio math
+    inputs.velocityRPM = (leaderVelocity.getValueAsDouble() * 60.0) / GEAR_RATIO;
+    inputs.appliedVolts = leaderAppliedVolts.getValueAsDouble();
+    inputs.currentAmps =
+        new double[] {leaderCurrent.getValueAsDouble(), followerCurrent.getValueAsDouble()}; //puts current values into an array as doubles
+  }
+
+  @Override
+  public void setVoltage(double volts) { //some method to set voltage for motor
+    leaderFlywheelMotor.setControl(new VoltageOut(volts));
+  }
+
+  @Override
+  public void setVelocity(double velocityRPM, double ffVolts) {//updating voltage to alter velocity of the motor
+    leaderFlywheelMotor.setControl(
+        new VelocityVoltage(
+            velocityRPM / 60.0,
+            0.0,
+            true,
+            ffVolts,
+            0,
+            false,
+            false,
+            false));
+  }
+
+  @Override
+  public void stop() { //stops the motor
+    leaderFlywheelMotor.stopMotor();
+  }
+
+  @Override
+  public void configurePID(double kP, double kI, double kD) { //sets the pid values
+    var config = new Slot0Configs();
```

Comment:
IMO, you guys should configure PID & FF within this implementation, as we have followed that design schema throughout the entire repo. In the future we may reevaluate, but for now id just configure it here.